ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -1428,6 +1418,56 @@ let message_destroy (_ : printer) rpc session_id params =
   in
   Client.Message.destroy_many ~rpc ~session_id ~messages
 
+let message_destroy_all (_ : printer) rpc session_id params =
+  let fail msg = raise (Cli_util.Cli_failure msg) in
+  let before = List.assoc_opt "before" params in
+  let after = List.assoc_opt "after" params in
+  let priority = List.assoc_opt "priority" params in
+  let before_date =
+    try Option.map Date.of_iso8601 before
```

Comment:
Can we change the accepted format to RFC3339 instead? or at least the message to users. We don't support ISO8601 because it's too wide of a format